tsdb/index/tsi1/index.go

```diff
@@ -885,6 +884,7 @@ func (i *Index) compactGroup(files []*IndexFile) {
 	n, err := IndexFiles(files).WriteTo(f)
 	if err != nil {
 		log.Printf("%s: error compacting index files: src=%s, path=%s, err=%s", IndexName, srcIDs, path, err)
+		f.Close()
 		return
 	}
```

Comment:
Changing to an explicit close here has two issues: 1) Any panic before this will cause `f` to not get closed, and 2) If someone else adds a another return in this function there's a decent chance that they could forget to add `f.Close()` which will cause issues. The `defer` is a catch all for any `return`.